bubbly/builtin/gen/gen.go

```diff
@@ -0,0 +1,182 @@
+// This package generates go structs for the builtin bubbly schema
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"os"
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/valocode/bubbly/bubbly/builtin"
+	"github.com/valocode/bubbly/store"
+	"github.com/zclconf/go-cty/cty"
+)
+
+const (
+	goSchemaFile    = "schema_gen.go"
+	goTablesFile    = "tables_gen.go"
+	packageName     = "builtin"
+	importStatement = `
+import (
+	"github.com/valocode/bubbly/api/core"
+	"github.com/zclconf/go-cty/cty"
+)	
+`
+)
+
+func genTablesFromSchema(graph *store.SchemaGraph) error {
+
+	var b bytes.Buffer
+	// Use this comment when testing to ignore compile errors...
+	// fmt.Fprintf(&b, "//+build ignore\n\n")
+	fmt.Fprintf(&b, "package %s\n\n", packageName)
+	fmt.Fprint(&b, importStatement)
+	fmt.Fprintf(&b, "var BuiltinTables = core.Tables{\n\n")
+	graph.Traverse(func(node *store.SchemaNode) error {
+		var (
+			table = node.Table
+			// tableName = camelToPascal(table.Name)
+		)
+		fmt.Fprintf(&b, "// #######################################\n")
+		fmt.Fprintf(&b, "// %s\n", strings.ToUpper(table.Name))
+		fmt.Fprintf(&b, "// #######################################\n")
+		fmt.Fprintf(&b, "\ttable(\"%s\",\n", table.Name)
+		fmt.Fprintf(&b, "\t\tfields(\n")
+		for _, field := range table.Fields {
+			fmt.Fprintf(&b, "\t\t\tfield(\"%s\", %s, %t),\n", field.Name, field.Type.GoString(), field.Unique)
+		}
+		fmt.Fprintf(&b, "\t\t),\n")
+		fmt.Fprintf(&b, "\t\tjoins(\n")
+		for _, join := range table.Joins {
+			fmt.Fprintf(&b, "\t\t\tjoin(\"%s\", %t, %t),\n", join.Table, join.Single, join.Unique)
+		}
+		fmt.Fprintf(&b, "\t\t),\n")
+		fmt.Fprintf(&b, "\t),\n")
```

Comment:
Would [text/template](https://golang.org/pkg/text/template/) not help with this kind of thing, @jlarfors?

Reply:
I did initially start using it, but then I was being less productive so I just moved to `Fprintf`. If this get's more complex/requires constant changing then maybe `text/template` could be introduced